api/src/main/java/org/openmrs/module/appointments/util/DateUtil.java

```diff
@@ -73,6 +73,16 @@ public static Date getEndOfDay() {
         calendar.set(Calendar.MILLISECOND, calendar.getMaximum(Calendar.MILLISECOND));
         return calendar.getTime();
     }
+
+    public static Date getStartOfDayByDate( Date date) {
+        Calendar calendar = Calendar.getInstance();
+        calendar.setTime(date);
+        calendar.set(Calendar.HOUR_OF_DAY, calendar.getMinimum(Calendar.HOUR_OF_DAY));
```

Comment:
I don't know about using calendar.getMinimum here.  I'd think there must be a good, existing method to do this either in core, or in apache commons, or whereever.  Maybe use `DateUtils.truncate` from apache commons...

Reply:
Just use `java.time` for this:

```java
LocalDate.now().atStartOfDay();
```

If we really need this as a date:

```java
Date.from(LocalDate.now().atStartOfDay().atZone(ZoneId.systemDefault()).toInstant());
```